roles/docker/tasks/debian.yml

```diff
@@ -1,26 +1,34 @@
 ---
+- name: Ensure old versions of Docker are not installed.
+  package:
+    name: '{{ item }}'
+    state: absent
+  with_items:
+    - docker
+    - docker-engine
 
 - name: install apt-transport-https
   apt:
     name: apt-transport-https
     state: installed
     update_cache: true
 
-- name: add docker repo
-  apt_repository:
-    repo: 'deb https://apt.dockerproject.org/repo debian-{{ansible_distribution_release}} main'
-    filename: docker
-    state: present
-
 - name: add docker key
   apt_key:
-    id: '58118E89F3A912897C070ADBF76221572C52609D'
+    id: '9DC858229FC7DD38854AE2D88D81803C0EBFCD88'
     data: "{{ lookup('file', 'docker.gpg') }}"
     state: present
 
+- name: add docker repo
+  apt_repository:
+    repo: 'deb [arch=amd64] {{debian_apt_repo}} {{ansible_distribution_release}} stable'
+    filename: docker
+    state: present
+    update_cache: true
+
 - name: install docker
   apt:
-    name: docker-engine
+    name: 'docker-ce={{debian_docker_version}}'
     state: installed
```

Comment:
Should this line be `state: present` or just deleting it as `present` is the default choice? According to Ansible official document, I don't think `installed` is the valid choice http://docs.ansible.com/ansible/latest/apt_module.html

Reply:
Good catch, it looks like `installed` was deprecated: https://github.com/ansible/ansible-modules-core/blob/devel/packaging/os/apt.py#L847-L851